src/index.js

```diff
@@ -0,0 +1,273 @@
+            ),
+            [t.arrowFunctionExpression(
+              [t.identifier("i")],
+              t.memberExpression(cssmodule, t.identifier("i"), true)
+            )]
+          ),
+          t.identifier("join"),
+          false
+        ),
+        [t.stringLiteral(" ")]
+      )
+    );
+  };
+
+  const updateClassName = (value, cssmodule) => {
+    if (t.isStringLiteral(value)) {
+      return value.replaceWith(computeClassName(value, cssmodule));
+    } else if (t.isIdentifier(value)) {
+      return value.replaceWith(computeClassName(value, cssmodule));
+    } else if (t.isCallExpression(value)) {
+      return replaceCallExpression(value, cssmodule);
+    } else {
+      console.log("TODO: updateClassName for %s", value.type);
+    }
+  };
+
+  /**
+   * Updates a JSX className value with the most appropriate CSS Module lookup.
+   *
+   * @param {Path} value <jsx className={value} />
+   * @param {Node<Identifier>} cssmodule cssmodule identifier
+   */
+  const updateJSXClassName = (value, cssmodule) => {
+    if (!t.isJSXExpressionContainer(value)) {
+      value.replaceWith({
+        type: "JSXExpressionContainer",
+        expression: value.node
+      });
+    }
+    updateClassName(value.get("expression"), cssmodule);
+  };
+
+  const buildRequire = template(`
+    const IMPORT_NAME = require(SOURCE);
+  `);
+
+  // TODO: template doesn't work for import.
+  const buildImport = ({IMPORT_NAME, SOURCE}) =>
+    t.importDeclaration(
+      [t.importDefaultSpecifier(IMPORT_NAME)], SOURCE
+    );
+
+  return {
+    visitor: {
+      JSXAttribute(path, state) {
+        if (path.get("name").node.name !== "className") {
+          return;
+        }
+
+        if (!state.cssModuleId) {
+          state.cssModuleId = path.scope.generateUidIdentifier(ROOT_CSSNAMES_IDENTIFIER);
+        }
+
+        updateJSXClassName(path.get("value"), state.cssModuleId);
+      },
+
+      CallExpression(path, state) {
+        const isCreateElementCall = matchesPatterns(
+          path.get("callee"),
+          ["React.createElement", "_react2.default.createElement"]
+        );
+
+        if (!isCreateElementCall) {
+          return;
+        }
+
+        if (!state.cssModuleId) {
+          state.cssModuleId = path.scope.generateUidIdentifier(ROOT_CSSNAMES_IDENTIFIER);
+        }
+
+        path.get("arguments")[1].get("properties").forEach((property) => {
+          if (property.node.key.name === "className") {
+            updateClassName(
+              property.get("value"),
+              state.cssModuleId
+            );
+          }
+        });
+      },
+
+      Program: {
+        enter(path, state:State) {
+          // detect if this is likely compiled source
+          if (path.scope.getBinding("_interopRequireDefault")) {
+            state.transformingOutput = true;
+            state.cssModuleId = path.scope.generateUidIdentifier(ROOT_CSSNAMES_IDENTIFIER);
+          }
+        },
+
+        exit(path, state:State) {
```

Comment:
Does this get called on exit of every file or on exit of the whole babel script?